src/comparator/wrappers.py

```diff
@@ -627,13 +627,20 @@ def oauth_scopes(self) -> Optional[Sequence[str]]:
             Sequence[str]: A sequence of OAuth scopes.
         """
         # Return the OAuth scopes, split on comma.
-        return tuple(
-            i.strip()
-            for i in self.service_pb.options.Extensions[client_pb2.oauth_scopes].split(
-                ","
-            )
-            if i
-        )
+        # fmt: off
+        oauth_scopes = []
+        for scope in self.service_pb.options.Extensions[client_pb2.oauth_scopes].split(","):
+            if scope:
+                oauth_scopes.append(
+                    WithLocation(
+                        scope.strip(),
+                        self.source_code_locations,
+                        self.path + (3, 1050),
```

Comment:
what is `(3, 1050)`?

Reply:
This path is for getting the source code line of the `oauth_scope`. The service option has field number 3, and `oauth_scope` option has field number 1050. So by path (...., 3, 1050), we can know the line number in the proto file where the `option(google.api.oauth_scopes) = ""` is defined.

Reply:
I see.